txscript/engine.go

```diff
@@ -555,6 +560,13 @@ func (vm *Engine) checkHashTypeEncoding(hashType SigHashType) error {
 // checkPubKeyEncoding returns whether or not the passed public key adheres to
 // the strict encoding requirements if enabled.
 func (vm *Engine) checkPubKeyEncoding(pubKey []byte) error {
+	if vm.hasFlag(ScriptVerifyWitnessPubKeyType) && vm.witness &&
```

Comment:
This looks like it would be a hard fork for non v0 segwit scripts.  The constraint is that all pubkeys must be compressed for all witness programs, not just v0.  However, the `vm.witness` flag is set to false on non v0 scripts in `verifyWitnessProgram`.